DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java

```diff
@@ -527,68 +528,70 @@ private String paramsTypesToString(Class<?>[] paramTypes) {
-                }
+            if (isDbArray(paramType) && candidateParamTypes[i].isArray()) {
+                paramType = convertDBArray(paramType, parameterizedTypes[i] == null ? null : parameterizedTypes[i][0]);
+            }
 
-                if (!isAssignableFrom(candidateParamTypes[i], paramType)) {
-                    acceptable = false;
-                    break;
-                }
+            boolean canAssignWithoutWidening = isAssignableFrom(candidateParamTypes[i], paramType);
+
+            Class<?> maybePrimitive = ClassUtils.wrapperToPrimitive(paramType);
+            boolean canAssignViaUnbox =
+                    maybePrimitive != null && isAssignableFrom(candidateParamTypes[i], maybePrimitive);
+            boolean canAssignWithWidening = maybePrimitive != null && candidateParamTypes[i].isPrimitive()
+                    && isWideningPrimitiveConversion(maybePrimitive, candidateParamTypes[i]);
+
+            if (!canAssignWithoutWidening && !canAssignViaUnbox && !canAssignWithWidening) {
+                return;
             }
+        }
 
-            // If the paramTypes includes 1+ varArgs check the classes match -- no need to check if there are 0 varArgs
-            if (candidate.isVarArgs() && paramTypes.length >= candidateParamTypes.length) {
-                Class<?> paramType = paramTypes[candidateParamTypes.length - 1];
+        // If the paramTypes includes 1+ varArgs check the classes match -- no need to check if there are 0 varArgs
+        if (candidate.isVarArgs() && paramTypes.length >= candidateParamTypes.length) {
+            Class<?> paramType = paramTypes[candidateParamTypes.length - 1];
 
-                if (isDbArray(paramType) && candidateParamTypes[candidateParamTypes.length - 1].isArray()) {
-                    paramType =
-                            convertDBArray(paramType, parameterizedTypes[candidateParamTypes.length - 1] == null ? null
-                                    : parameterizedTypes[candidateParamTypes.length - 1][0]);
-                }
+            if (isDbArray(paramType) && candidateParamTypes[candidateParamTypes.length - 1].isArray()) {
+                paramType = convertDBArray(paramType, parameterizedTypes[candidateParamTypes.length - 1] == null ? null
+                        : parameterizedTypes[candidateParamTypes.length - 1][0]);
+            }
 
-                if (candidateParamTypes.length == paramTypes.length && paramType.isArray()) {
-                    if (!isAssignableFrom(candidateParamTypes[candidateParamTypes.length - 1], paramType)) {
-                        acceptable = false;
-                    }
-                } else {
-                    final Class<?> lastClass = candidateParamTypes[candidateParamTypes.length - 1].getComponentType();
+            if (candidateParamTypes.length == paramTypes.length && paramType.isArray()) {
+                if (!isAssignableFrom(candidateParamTypes[candidateParamTypes.length - 1], paramType)) {
```

Comment:
I suspect I should move the new logic into a helper, and allow unboxing and widening on varargs, too.